### PR TITLE
add support for token based access

### PR DESCRIPTION
**Pull request by jafferli**
add support for token based access api

### CODE REVIEW

File: pygraylog/graylogapi.py [hunk 1]
```diff
@@ -4,6 +4,11 @@
 from urlparse import urlparse
 class GraylogAPI(object):
     def __init__(self, url, username=None, password=None, api_key=None):
+
+        # check if the username / password vs. api_key cannot be provided at same time
+        if username is not None and api_key is not None:
```
Comment (zmallen) [inline pygraylog/graylogapi.py:9]: This check is raising an error in the unit tests. My question is: should it not matter if the username, password and api_key are set? We could just prefer the API key if it is provided, which is what seems like you added on line 42.

File: pygraylog/graylogapi.py [hunk 2]
```diff
@@ -31,7 +36,13 @@ def method(**kwargs):
             return self._(name)
 
     def build_auth_header(self):
-        payload = self.username + ':' + self.password
+        
+        # use the api_key if it is been provided
+        if self.api_key is not None:
+            payload = self.api_key + ':token'
```
Comment (zmallen) [inline pygraylog/graylogapi.py:42]: 👍  this looks like the way Graylog constructs the token header, http://docs.graylog.org/en/2.2/pages/configuration/rest_api.html

**Review by zmallen — CHANGES_REQUESTED** [review]
I think the authorization header looks correct, and it matches documentation. Thanks for the P/R!

Could you please add some unit tests and fix the current broken unit test? It throws an exception since we generate a Graylog Object w/ username/password & an API key. I would split that unit test into a username/password one and a token based one. Lastly, could you increment `setup.py` version of pygraylog so I can do a pypi release?

Thank you!

**Comment by zmallen** [comment]
One last observation - update readme to show token based auth as well!

**Comment by jafferli** [comment]
update the patch-1 branch for
1) setup.py for version change to 0.3.2
2) unit test item

**Review by zmallen — APPROVED** [review]

**Comment by zmallen** [comment]
Thank you @jafferli ! I'll get this pushed to pypi asap

**Pull request merged**